### PR TITLE
feat: Handle dependent feature toggles

### PR DESCRIPTION
**Pull request by chriswk**
- A new feature added in beta to Unleash 5.6. With dependent feature toggle you can configure parent-child relationships. This change makes the SDK evaluate according to the client specifications defined for this.

### CODE REVIEW

File: src/main/java/io/getunleash/DefaultUnleash.java [hunk 1]
```diff
@@ -168,40 +169,89 @@ private FeatureEvaluationResult getFeatureEvaluationResult(
...
-            for (ActivationStrategy strategy : featureToggle.getStrategies()) {
-                Strategy configuredStrategy = getStrategy(strategy.getName());
-                if (configuredStrategy == UNKNOWN_STRATEGY) {
-                    LOGGER.warn(
-                            "Unable to find matching strategy for toggle:{} strategy:{}",
-                            toggleName,
-                            strategy.getName());
-                }
+            // Dependent toggles, no point in evaluating child strategies if our dependencies are
+            // not enabled
```
Comment (kwasniew) [inline src/main/java/io/getunleash/DefaultUnleash.java:177]: not enabled -> not satisfied. satisfied may mean disabled or enabled depending on how we configure our dependencies

File: src/test/java/io/getunleash/DefaultUnleashTest.java [hunk 2]
```diff
@@ -275,4 +272,18 @@ public void client_identifier_handles_api_key_being_null() {
         assertThat(id)
                 .isEqualTo("f83eb743f4c8dc41294aafb96f454763e5a90b96db8b7040ddc505d636bdb243");
     }
+
+    /*    @Test
```
Comment (kwasniew) [inline src/test/java/io/getunleash/DefaultUnleashTest.java:276]: what's up with this test?

**Review by kwasniew — APPROVED** [review]
Very nice. A few minor comments inside. 2 things I'd test since they are not part of the client specification:
* metrics are not called on dependent features
* impression events are called on dependent features for easier debugging

File: src/main/java/io/getunleash/DefaultUnleash.java [hunk 3]
```diff
@@ -168,40 +169,89 @@ private FeatureEvaluationResult getFeatureEvaluationResult(
...
-                    result.setVariant(variant);
-                    return result;
                 }
             }
             return new FeatureEvaluationResult(false, defaultVariant);
         }
     }
 
+    //
+    // Checks a feature's dependencies (if it has any
```
[the remark this answers is not in the document]
Comment (chriswk) [reply src/main/java/io/getunleash/DefaultUnleash.java:213]: yeah. thanks

File: src/test/java/io/getunleash/DefaultUnleashTest.java (hunk 2 above)
Comment (chriswk) [reply src/test/java/io/getunleash/DefaultUnleashTest.java:276]: It isn't needed, so I'll remove it

**Review by andreas-unleash — APPROVED** [review]
LGTM

**Pull request merged**